File: Sorting/In C/ShellSort.c
#include<stdio.h>

void getArray(int array[],int n){

    for(int i=0;i<n;i++){
        scanf("%d",&array[i]);
    }
}

void dispArray(int array[],int n){

    for(int i=0;i<n;i++){
        printf("%d ",array[i]);
    }   
}

void ShellSort(int array[],int n){
    int temp=0;
        for(int i=0;i<n;i++){
            for(int j=1;j<=(n-1-i);j++){
                if(array[j-1]>array[j]){
                    temp=array[j];
                    array[j]=array[j-1];
                    array[j-1]=temp;
                }
            }
            printf("%s",array);
        }
    }

int main(){

    int n,array[50],key,r;
    printf("Enter number of elements in the array: ");
    scanf("%d",&n);
    getArray(array,n);

    printf("The given array is array is ");
    dispArray(array,n);
    ShellSort(array,n);
    printf("\nThe Sorted array is array is ");
    dispArray(array,n);

}

File: Sorting/In Python #prefer/ShellSort.py
#Shell Sort

# next level of insertion sort  //Larger swap, same methodology
# Note: double swap allowed for same variable

def shell(array,size):

    # find largest swap interval value in the array
    span=size//2
    
    while(span>0):
    
        for i in range(span,size):
            j=i
            # store a[i] in temporary varible and swap assign till possible
            temp=array[i]  
            # there is no point in the while loop if j<span LOL                                                     
            while(array[j-span]>temp and j>=span):                              
                array[j]=array[j-span]
                j-=span
                print("j=",j)

            # assign temp to the last swapp assigned position j
            array[j]=temp  
        # reduce swap interval to half slowly
        span=span//2
        
    return array
   
def getList(array,n):
    for i in range(n):
        print("Enter value {} :".format(i+1))
        val=int(input())
        array.append(val)

n=int(input("\nEnter the number of elements in the list:"))

print("\nEnter the list elements:")
array=[]
getList(array,n)

print("\nThe sorted array is ")
print(array)
